### PR TITLE
Extinguished burning clothes no longer burn forever.

### LINKED ISSUE #42800: On-Fire Sprites for clothing and Hood Toggle Stay on Fire
## Round ID: 101883

Combat happened, someone caught on fire (gun), then I caught on fire (proximity). Took the necessary steps to rectify that but noticed my clothes were still showing that they were on fire. I took a shower but to no avail. The sprites appear to get stuck on their on-fire variants and it appears to extend to associated sprites as well (such as the hood toggle button for the HUD). This is the first time this has happened thus the bug report. 
Image Related:
autism station fire fire fire

The image doesn't show the HUD button on fire, but after I took the snapshot, it was. This might have occurred after a toggle.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Fixes #42800 as far as I'm aware.
So from what I understand, clothes sprites were working when they extinguished  immediately after catching fire. But, when your clothes catch fire and get the damage overlay as well, they weren't losing the burning overlay, ever. Solution may be a bit hacky, but it seems to work in all cases, including throwing burning clothes onto showers, and individually extinguishing burning clothes. 

## Why It's Good For The Game

Bugs bad.

## Changelog
:cl:
fix: Burning clothes no longer burn forever if you're set on fire for too long.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -188,6 +189,13 @@
 	if(method == TOUCH)
 		M.adjust_fire_stacks(-(reac_volume / 10))
 		M.ExtinguishMob()
+		if(istype(M, /mob/living/carbon))
+			var/mob/living/carbon/C = M
+			for(var/X in C.get_equipped_items())
+				var/obj/item/I = X
+				I.acid_level = 0
+				I.extinguish()
+				I.cut_overlay(GLOB.fire_overlay, TRUE)
```
Comment (Dennok) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: ExtinguishMob() call it. 
add I.cut_overlay(GLOB.fire_overlay, TRUE) in ExtinguishMob()
Or better just `update_icon()`

https://github.com/tgstation/tgstation/blob/85379a5269963b2463f7aa04a12e592cbe023ed3/code/modules/mob/living/carbon/carbon.dm#L853-L858

File: code/game/objects/obj_defense.dm [hunk 2]
```diff
@@ -224,6 +224,7 @@ GLOBAL_DATUM_INIT(acid_overlay, /mutable_appearance, mutable_appearance('icons/e
 	if(resistance_flags & ON_FIRE)
 		resistance_flags &= ~ON_FIRE
 		cut_overlay(GLOB.fire_overlay, TRUE)
+		update_icon_state()
```
Comment (Dennok) [inline code/game/objects/obj_defense.dm]: Why not `update_icon()` ?

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -163,6 +163,7 @@
 /datum/reagent/water/reaction_obj(obj/O, reac_volume)
 	O.extinguish()
 	O.acid_level = 0
+	O.cut_overlay(GLOB.fire_overlay, TRUE)
```
Comment (Dennok) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: Why not `update_icon()` ?

**Comment by Pumpkinoe** [comment]
Will this also fix acid forever being on clothes?

File: code/modules/mob/living/carbon/carbon.dm [hunk 4]
```diff
@@ -855,6 +855,7 @@
 		var/obj/item/I = X
 		I.acid_level = 0 //washes off the acid on our clothes
 		I.extinguish() //extinguishes our clothes
+		I.cut_overlay(GLOB.fire_overlay, TRUE)
```
Comment (Dennok) [inline code/modules/mob/living/carbon/carbon.dm]: .extinguish() already call cut_overlay(GLOB.fire_overlay, TRUE)
It really need be doubled here?

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 5]
```diff
@@ -163,6 +163,7 @@
 /datum/reagent/water/reaction_obj(obj/O, reac_volume)
 	O.extinguish()
 	O.acid_level = 0
+	O.update_icon()
```
Comment (Dennok) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: .extinguish() now call update_icon()
It really need be doubled here?

File: code/modules/mob/living/carbon/carbon.dm (hunk 4 above)
Comment (ArcaneMusic) [reply code/modules/mob/living/carbon/carbon.dm]: This one does need to be doubled. Without this cut_overlay, the fire overlay stays forever when there's mixed overlays (Acid, Fire, and/or Damage overlays).

**Pull request merged**